components/script/dom/node.rs

```diff
@@ -461,19 +462,24 @@ impl Node {
     }
 
     /// Iterates over this node and all its descendants, in preorder.
-    pub fn traverse_preorder(&self) -> TreeIterator {
-        TreeIterator::new(self)
+    pub fn traverse_preorder<T>(&self) -> TreeIterator<T>
+        where T: DerivedFrom<Node> {
+        TreeIterator::<T>::new(self)
     }
 
-    pub fn inclusively_following_siblings(&self) -> NodeSiblingIterator {
-        NodeSiblingIterator {
+    pub fn inclusively_following_siblings<T>(&self) -> SiblingIterator<T>
+        where T: DerivedFrom<Node> {
```

Comment:
Nit: the opening brace must be on its own line if there is a indented `where` clause.